client/src/components/EditProfile/EditProfile.tsx

```diff
@@ -6,46 +7,129 @@ import CardContent from '@material-ui/core/CardContent';
 import CircularProgress from '@material-ui/core/CircularProgress';
 import Button from '@material-ui/core/Button';
 import Box from '@material-ui/core/Box';
-import { Formik } from 'formik';
+import { Formik, FormikHelpers } from 'formik';
 import * as Yup from 'yup';
+import { format } from 'date-fns';
 
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { useAuth } from '../../context/useAuthContext';
 import useStyles from './useStyles';
+import { IUseUser } from '../../context/useUserContext';
+import { updateAuthFields, createOrUpdateProfileFields } from '../../helpers/APICalls/profileFields';
 
 interface FormValues {
   firstName: string;
   lastName: string;
   email: string;
   phone: string;
-  whereYouLive: string;
-  describeYourself: string;
+  city: string;
+  description: string;
+  startDate: string;
+  endDate: string;
+  priceRate: string;
 }
 
-const EditProfile = (): JSX.Element => {
+interface Props extends IUseUser {
+  handleChangedAnything: () => void;
+  handleChangedAnythingToFalse: () => void;
+}
+
+const EditProfile: FC<Props> = (props) => {
   const classes = useStyles();
-  const handleSubmit = (values: FormValues): void => {
-    // handle form values;
-    console.log(values);
+  const { userState, dispatchUserContext, handleChangedAnything, handleChangedAnythingToFalse } = props;
+  const { loggedInUser } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const handleSubmit = (values: FormValues, { setSubmitting }: FormikHelpers<FormValues>): void => {
+    // calling this function will not display confirm box to user
+    handleChangedAnythingToFalse();
+    let authFieldsChange = true;
+    let otherFieldsChange = true;
+    const { firstName, lastName, ...otherValues } = values;
+    // if any fields has changes only then we have to fire a request
+    if (loggedInUser?.firstName === firstName && loggedInUser?.lastName === lastName) {
+      authFieldsChange = false;
+    }
+    // if any fields has changes only then we have to fire a request
+    if (
+      userState.phone === otherValues.phone &&

```

Comment:
i think this logic can be rewritten so that it's more maintainable something like

```
for (let key in userState) {
  if (userState[key] !== otherValues[key]) {
    otherFieldsChange = false;

    return;
  }
}
```

Reply:
yes, you are right. I will do that